sql/core/src/test/scala/org/apache/spark/sql/execution/vectorized/ColumnarBatchSuite.scala

```diff
@@ -1248,4 +1251,238 @@ class ColumnarBatchSuite extends SparkFunSuite {
+    nullRow.setNullAt(0)
+    columnBuilder.appendFrom(nullRow, 0)
+    for (i <- 1 until 16) {
+      row.setInt(0, i)
+      columnBuilder.appendFrom(row, 0)
+    }
+
+    val column = new CachedBatchColumnVector(
+      JavaUtils.bufferToArray(columnBuilder.build), 1024, dataType)
+
+    assert(column.isNullAt(0) == true)
+    for (i <- 1 until 16) {
+      assert(column.isNullAt(i) == false)
+      assert(column.getInt(i) == i)
+    }
+    column.close
+  }
+
+  test("CachedBatch long Apis") {
+    val dataType = LongType
+    val columnBuilder = ColumnBuilderHelper(dataType, 1024, "col", true)
+    val row = new SpecificInternalRow(Array(dataType))
+
+    val nullRow = InternalRow(false)
+    nullRow.setNullAt(0)
+    columnBuilder.appendFrom(nullRow, 0)
+    for (i <- 1 until 16) {
+      row.setLong(0, i.toLong)
+      columnBuilder.appendFrom(row, 0)
+    }
+
+    val column = new CachedBatchColumnVector(
+      JavaUtils.bufferToArray(columnBuilder.build), 1024, dataType)
+
+    assert(column.isNullAt(0) == true)
+    for (i <- 1 until 16) {
+      assert(column.isNullAt(i) == false)
+      assert(column.getLong(i) == i.toLong)
+    }
+    column.close
+  }
+
+  test("CachedBatch float Apis") {
+    val dataType = FloatType
+    val columnBuilder = ColumnBuilderHelper(dataType, 1024, "col", true)
+    val row = new SpecificInternalRow(Array(dataType))
+
+    val nullRow = InternalRow(false)
+    nullRow.setNullAt(0)
+    columnBuilder.appendFrom(nullRow, 0)
+    for (i <- 1 until 16) {
+      row.setFloat(0, i.toFloat)
+      columnBuilder.appendFrom(row, 0)
+    }
+
+    val column = new CachedBatchColumnVector(
+      JavaUtils.bufferToArray(columnBuilder.build), 1024, dataType)
+
+    assert(column.isNullAt(0) == true)
+    for (i <- 1 until 16) {
+      assert(column.isNullAt(i) == false)
+      assert(column.getFloat(i) == i.toFloat)
+    }
+    column.close
+  }
+
+  test("CachedBatch double Apis") {
+    val dataType = DoubleType
+    val columnBuilder = ColumnBuilderHelper(dataType, 1024, "col", true)
+    val row = new SpecificInternalRow(Array(dataType))
+
+    val nullRow = InternalRow(false)
+    nullRow.setNullAt(0)
+    columnBuilder.appendFrom(nullRow, 0)
+    for (i <- 1 until 16) {
+      row.setDouble(0, i.toDouble)
+      columnBuilder.appendFrom(row, 0)
+    }
+
+    val column = new CachedBatchColumnVector(
+      JavaUtils.bufferToArray(columnBuilder.build), 1024, dataType)
+
+    assert(column.isNullAt(0) == true)
+    for (i <- 1 until 16) {
+      assert(column.isNullAt(i) == false)
+      assert(column.getDouble(i) == i.toDouble)
+    }
+    column.close
+  }
+
+  test("CachedBatch String type Apis") {
+    val dataType = StringType
+    val columnBuilder = ColumnBuilderHelper(dataType, 1024, "col", true)
+
+    val nullRow = InternalRow("")
+    nullRow.setNullAt(0)
+    columnBuilder.appendFrom(nullRow, 0)
+    for (i <- 1 until 16) {
+      val converter = UnsafeProjection.create(Array[DataType](dataType))
```

Comment:
Do we need this?
How about simply like the following with `SpecificInternalRow`?

```
row.update(0, UTF8String.fromString(...))
```